File: php/max-depth-binary-tree.php
<?php

class Node {
    public $data;
    public $left;
    public $right;

    public function __construct($data) {
        $this->data = $data;
        $this->left = null;
        $this->right = null;
    }
}

class Tree {
    public $root;
}

class Solution {
    public static function maxDepth($root) {
        if (is_null($root)) {
            return 0;
        }

        $leftDepth = self::maxDepth($root->left);
        $rightDepth = self::maxDepth($root->right);

        return max($leftDepth, $rightDepth) + 1;
    }
}

$tree = new Tree();
$tree->root = new Node(3);
$tree->root->left = new Node(9);
$tree->root->right = new Node(20);
$tree->root->right->left = new Node(15);
$tree->root->right->right = new Node(7);

echo Solution::maxDepth($tree->root); // 3

File: python/max-depth-binary-tree.py
class Node:
    def __init__(self, val):
        self.val = val
        self.left = None
        self.right = None

class Tree:
    def __init__(self, root):
        self.root = root

class Solution:
    def maxDepth(self, root):
        if root is None:
            return 0
        return max(self.maxDepth(root.left), self.maxDepth(root.right))+1
    
tree = Tree(Node(1))
tree.root.left = Node(2)
tree.root.right = Node(3)
tree.root.left.left = Node(4)
tree.root.left.left.left = Node(5)

print(Solution().maxDepth(tree.root))